src/UnreadableStreamException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ancarda\File;
+
+use Exception;
+
+class UnreadableStreamException extends Exception
```

Comment:
Could you change this to extend from `InvalidArgumentException` rather than the base SPL `Exception` class? That is thrown elsewhere in `Detector` for bad inputs and I think of Unreadable Stream as an example of an Invalid Argument.